Source/AwsCommonRuntimeKit/auth/credentials/CRTCredentialsProviderCachedConfig.swift

```diff
@@ -2,8 +2,8 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 public protocol CRTCredentialsProviderCachedConfig {
-    var shutDownOptions: CRTCredentialsProviderShutdownOptions? {get set}
-    var source: CRTAWSCredentialsProvider {get set}
+    var shutdownCallback: ShutdownCallback? { get }
+    var source: CRTAWSCredentialsProvider { get }
     /// refresh time in ms
-    var refreshTime: Int64 {get set}
+    var refreshTime: Int64 { get }
```

Comment:
it would have been nice if `refreshTime` was `TimeInterval`.

Reply:
Sure, I will update it. Thanks.

Reply:
Fixed, please let me know if you have other ideas to improve the APIs.